consensus/spos/worker.go

```diff
@@ -381,6 +420,56 @@ func (wrk *Worker) Extend(subroundId int) {
 	log.Debug(fmt.Sprintf("account state is reverted to snapshot\n"))
 
 	wrk.blockProcessor.RevertAccountState()
+
+	shouldBroadcastLastCommittedBlock := wrk.consensusState.IsSelfLeaderInCurrentRound() &&
+		wrk.consensusService.IsSubroundSignature(subroundId)
+
+	if shouldBroadcastLastCommittedBlock {
+		wrk.broadcastLastCommittedBlock()
+	}
+
+	wrk.dysplaySignatureStatistic()
+
+	wrk.mutHashConsensusMessage.Lock()
+	wrk.mapHashConsensusMessage = make(map[string][]*consensus.Message)
+	wrk.mutHashConsensusMessage.Unlock()
+}
+
+func (wrk *Worker) broadcastLastCommittedBlock() {
+	header := wrk.blockChain.GetCurrentBlockHeader()
+	body := wrk.blockChain.GetCurrentBlockBody()
+
+	if header == nil || header.IsInterfaceNil() {
+		return
+	}
+
+	// broadcast block body and header
+	err := wrk.broadcastMessenger.BroadcastBlock(body, header)
+	if err != nil {
+		log.Error(err.Error())
+	}
+
+	// broadcast header to metachain
+	err = wrk.broadcastMessenger.BroadcastHeader(header)
+	if err != nil {
+		log.Error(err.Error())
+	}
+}
+
+func (wrk *Worker) dysplaySignatureStatistic() {
+	wrk.mutHashConsensusMessage.RLock()
+	for hash, consensusMessages := range wrk.mapHashConsensusMessage {
+		log.Info(fmt.Sprintf("proposed header with hash %s has received %d signatures\n",
```

Comment:
do we really need all this information ? maybe set to log.Debug, and launch amazon nodes in debug mode.

Reply:
Actually this loop would print almost all the time only one line and it would be only when a proposed block didn't achieved the minimum necessary number of signatures (2/3 + 1) and the round would be dropped from signature round. But it will help a lot to know how big is the majority for a proposed block.